src/liger_kernel/ops/experimental/gemm_split_k_fp8_e4m3.py

```diff
@@ -0,0 +1,302 @@
+    group_id = pid // width
+    pid_in_group = pid % width
+    group_size = tl.minimum(grid_m - group_id * group_m, group_m)
+    pid_m = group_id * group_m + (pid_in_group % group_size)
+    pid_n = pid_in_group // group_size
+
+    return pid_m, pid_n
+
+
+@triton.jit
+def gemm_split_k_kernel_forward(
+    a_ptr,
+    b_ptr,
+    c_ptr,
+    stride_am,
+    stride_ak,
+    stride_bk,
+    stride_bn,
+    stride_cm,
+    stride_cn,
+    m,
+    n,
+    k,
+    block_m: tl.constexpr,
+    block_n: tl.constexpr,
+    block_k: tl.constexpr,
+    split_k: tl.constexpr,
+    group_m: tl.constexpr,
+):
+    pid = tl.program_id(0)
+    pid_k = tl.program_id(1)
+    grid_k = tl.cdiv(k, block_k * split_k)
+
+    pid_m, pid_n = grouped_launch(pid, m, n, block_m, block_n, group_m)
+
+    offs_m = tl.multiple_of(pid_m * block_m + tl.arange(0, block_m), block_m)
+    offs_n = tl.multiple_of(pid_n * block_n + tl.arange(0, block_n), block_n)
+    offs_k = pid_k * block_k + tl.arange(0, block_k)
+
+    offs_am = tl.max_contiguous(tl.multiple_of(offs_m, block_m), block_m)
+    offs_bn = tl.max_contiguous(tl.multiple_of(offs_n, block_n), block_n)
+
+    a_ptrs = a_ptr + (offs_am[:, None] * stride_am + offs_k[None, :] * stride_ak)
+    b_ptrs = b_ptr + (offs_k[:, None] * stride_bk + offs_bn[None, :] * stride_bn)
+
+    acc = tl.zeros((block_m, block_n), dtype=tl.float32)
+
+    for k_ in range(0, grid_k, step=2):
+        k_remaining = k - k_ * (block_k * split_k)
+
+        mask_a = offs_k[None, :] < k_remaining
+        mask_b = offs_k[:, None] < k_remaining
+
+        a = tl.load(a_ptrs, mask=mask_a, other=0.0)
+        b = tl.load(b_ptrs, mask=mask_b, other=0.0)
+
+        # fp8 input dot product (supported types: [fp8e4nv, fp8e5, fp8e4b15])
+        acc = tl.dot(a, b, acc)
+
+        a_ptrs += block_k * split_k * stride_ak
+        b_ptrs += block_k * split_k * stride_bk
+
+    # cast to fp16 pre-store
+    acc = acc.to(tl.float16)
+
+    offs_m = pid_m * block_m + tl.arange(0, block_m)
+    offs_n = pid_n * block_n + tl.arange(0, block_n)
+
+    c_ptrs = c_ptr + (offs_m[:, None] * stride_cm + offs_n[None, :] * stride_cn)
+    mask = (offs_m < m)[:, None] & (offs_n < n)[None, :]
+
+    tl.atomic_add(c_ptrs, acc, mask=mask)
+
+
+def forward_kernel(ctx, a, b):
+    ctx.save_for_backward(a, b)
+
+    # cast to FP8
+    # structure:
+    #   | 1 bit sign | 4 bit exponent | 3 bit mantissa |
+    a, b = a.to(torch.float8_e4m3fn), b.to(torch.float8_e4m3fn)
+
+    m, k = a.shape
+    _, n = b.shape
+
+    block_m, block_n, block_k, num_stages, num_warps, split_k, group_m = (
+        calculate_gemm_settings(m, n, k)
+    )
+
+    total_blocks_m = triton.cdiv(m, block_m)
+    total_blocks_n = triton.cdiv(n, block_n)
+    total_programs_mn = total_blocks_m * total_blocks_n
+    total_programs_k = split_k
+
+    grid = (total_programs_mn, total_programs_k)
+
+    c = torch.zeros((m, n), device=a.device, dtype=torch.float16)
+    gemm_split_k_kernel_forward[grid](
```

Comment:
Could you add some comments to explain the reason of using split_k implementation such as in which scenario it's preferred?